model/configuration.py

```diff
@@ -1171,7 +1171,12 @@ def __get__(self, owner_instance, owner_type):
                 "owner must be an instance of ConfigurationSettingsMetadataOwner type"
             )
 
-        return owner_instance.get_setting_value(self._key)
+        setting_value = owner_instance.get_setting_value(self._key)
+
+        if setting_value is None:
+            setting_value = self.default
```

Comment:
Are there any settings for which null/None is a useful/used value AND the default is something else? I recall changing the default audiobook vendor/distributor exclusion when we enabled support for OD abooks last year because it was defaulting to OverDrive, if not set; but don't remember the details.

Reply:
I thought about it. The most obvious option would be boolean values but they're usually represented as dropdown lists. For example, [this SAML configuration setting](https://github.com/ThePalaceProject/circulation/blob/main/api/saml/configuration/model.py#L71):
```python
patron_id_use_name_id = ConfigurationMetadata(
    key="saml_patron_id_use_name_id",
    label=_("Patron ID: SAML NameID"),
    description=_(
        "Configuration setting indicating whether SAML NameID should be searched for a unique patron ID. "
        "If NameID found, it will supersede any SAML attributes selected in the next section."
    ),
    type=ConfigurationAttributeType.SELECT,
    required=False,
    default="true",
    options=[
        ConfigurationOption("true", "Use SAML NameID"),
        ConfigurationOption("false", "Do NOT use SAML NameID"),
    ],
)
```